lib/services/cordova-project-plugins-service.ts

```diff
@@ -0,0 +1,682 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import {EOL} from "os";
+import * as util from "util";
+import * as pluginsDataLib from "../plugins-data";
+import Future = require("fibers/future");
+import * as helpers from "../common/helpers";
+import semver = require("semver");
+import * as cordovaPlugins from "./cordova-plugins";
+import * as validUrl from "valid-url";
+
+export class CordovaProjectPluginsService implements IPluginsService {
+	private static CORE_PLUGINS_PROPERTY_NAME = "CorePlugins";
+	private static CORDOVA_PLUGIN_VARIABLES_PROPERTY_NAME = "CordovaPluginVariables";
+	private static MESSAGES = ["Core Plugins", "Advanced Plugins", "Marketplace Plugins"];
+	private _identifierToPlugin: IDictionary<IPlugin>;
+	private get identifierToPlugin(): IDictionary<IPlugin> {
+		if (!this._identifierToPlugin) {
+			this.loadPluginsData().wait();
+		}
+
+		return this._identifierToPlugin;
+	}
+
+	private pluginsForbiddenConfigurations: IStringDictionary = {
+		"com.telerik.LivePatch": this.$projectConstants.DEBUG_CONFIGURATION_NAME
+	};
+
+	constructor(private $cordovaPluginsService: cordovaPlugins.CordovaPluginsService,
+		private $marketplacePluginsService: ICordovaPluginsService,
+		private $errors: IErrors,
+		private $logger: ILogger,
+		private $project: Project.IProject,
+		private $prompter: IPrompter,
+		private $loginManager: ILoginManager,
+		private $projectConstants: Project.IProjectConstants,
+		private $options: IOptions,
+		private $fs: IFileSystem) { }
+
+	private loadPluginsData(): IFuture<void> {
+		return (() => {
+			// Cordova plugin commands are only applicable to Cordova projects
+			this.$project.ensureCordovaProject();
+			this.$loginManager.ensureLoggedIn().wait();
+			this._identifierToPlugin = Object.create(null);
+			Future.wait([this.createPluginsData(this.$cordovaPluginsService),
+				this.createPluginsData(this.$marketplacePluginsService)]);
+		}).future<void>()();
+	}
+
+	public findPlugins(keywords: string[]): IFuture<IBasicPluginInformation[]> {
+		return (() => {
+			return this.$cordovaPluginsService.getPlugins(keywords);
+		}).future<IBasicPluginInformation[]>()();
+	}
+
+	public fetch(pluginIdentifiers: string[]): IFuture<void> {
```

Comment:
This method was never intended to be used with multiple pluginIdentifiers. 
You can consider either changing the method's signature to accept only **one** pluginIdentifier, or refactor the method to actually be able to install multiple plugins.
My personal choice would be the first option as the second option would require more testing and refactoring for a functionality we do not yet need.